tests/forms_tests/field_tests/test_choicefield.py

```diff
@@ -82,10 +82,49 @@ def test_choicefield_disabled(self):
         f = ChoiceField(choices=[('J', 'John'), ('P', 'Paul')], disabled=True)
         self.assertWidgetRendersTo(
             f,
-            '<select id="id_f" name="f" disabled required><option value="J">John</option>'
+            '<select id="id_f" name="f" disabled><option value="J">John</option>'
             '<option value="P">Paul</option></select>'
         )
 
+    def test_choicefield_doesnt_render_required_when_impossible_to_select_empty_field(self):
+        f = ChoiceField(choices=[('J', 'John'), ('P', 'Paul')])
+        self.assertWidgetRendersTo(
+            f,
```

Comment:
Despite the existing style of the first test, I would remove the intermediate `f` variable in the new tests as it'll help balance line lengths and make things more readable.